numpy 1.23 compatibility

**Describe the bug**
We appear to have some fancy-slicing errors with numpy 1.23.  These arise entirely from the new requirement that iterable indices must be tuples (not lists).

For example, this test currently fails on numpy 1.23:

```python
――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――― test_sync_multi[2-0-None] ―――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――

aggregate = None, ndim = 2, axis = 0

    @pytest.mark.parametrize("aggregate", [None, np.mean, np.sum])
    @pytest.mark.parametrize(
        "ndim,axis", [(1, 0), (1, -1), (2, 0), (2, 1), (2, -1), (3, 0), (3, 2), (3, -1), (4, 0), (4, 3), (4, -1)]
    )
    def test_sync_multi(aggregate, ndim, axis):
        data = np.ones([6] * ndim, dtype=np.float)
    
        # Make some slices that don't fill the entire dimension
        slices = [slice(1, 3), slice(3, 4)]
        dsync = librosa.util.sync(data, slices, aggregate=aggregate, axis=axis)
    
        # Check the axis shapes
        assert dsync.shape[axis] == len(slices)
    
        s_test = list(dsync.shape)
        del s_test[axis]
        s_orig = list(data.shape)
        del s_orig[axis]
        assert s_test == s_orig
    
        # The first slice will sum to 2 and have mean 1
        idx = [slice(None)] * ndim
        idx[axis] = 0
        if aggregate is np.sum:
            assert np.allclose(dsync[idx], 2)
        else:
>           assert np.allclose(dsync[idx], 1)
E           IndexError: only integers, slices (`:`), ellipsis (`...`), numpy.newaxis (`None`) and integer or boolean arrays are valid indices

tests/test_multichannel.py:70: IndexError

 tests/test_multichannel.py ⨯                             
```
but it would be fixed by saying `dsync[tuple(idx)]`.

I've audited the codebase and all issues are confined to tests (not the library itself), so this should be an easy fix.  PR forthcoming.

--- a/tests/test_core.py
+++ b/tests/test_core.py
@@ -2122,6 +2122,8 @@ def test_pcen_stream_multi(axis):
     slice2 = [slice(None)] * x.ndim
     slice2[axis] = slice(10, None)
 
+    slice1 = tuple(slice1)
+    slice2 = tuple(slice2)
     # Compute pcen piecewise
     p1, zf1 = librosa.pcen(x[slice1], return_zf=True, axis=axis)
     p2, zf2 = librosa.pcen(x[slice2], zi=zf1, return_zf=True, axis=axis)

--- a/tests/test_multichannel.py
+++ b/tests/test_multichannel.py
@@ -65,13 +65,13 @@ def test_sync_multi(aggregate, ndim, axis):
     idx = [slice(None)] * ndim
     idx[axis] = 0
     if aggregate is np.sum:
-        assert np.allclose(dsync[idx], 2)
+        assert np.allclose(dsync[tuple(idx)], 2)
     else:
-        assert np.allclose(dsync[idx], 1)
+        assert np.allclose(dsync[tuple(idx)], 1)
 
     # The second slice will sum to 1 and have mean 1
     idx[axis] = 1
-    assert np.allclose(dsync[idx], 1)
+    assert np.allclose(dsync[tuple(idx)], 1)
 
 
 def test_stft_multi(y_multi):

--- a/tests/test_util.py
+++ b/tests/test_util.py
@@ -877,13 +877,13 @@ def test_sync(aggregate, ndim, axis):
     idx = [slice(None)] * ndim
     idx[axis] = 0
     if aggregate is np.sum:
-        assert np.allclose(dsync[idx], 2)
+        assert np.allclose(dsync[tuple(idx)], 2)
     else:
-        assert np.allclose(dsync[idx], 1)
+        assert np.allclose(dsync[tuple(idx)], 1)
 
     # The second slice will sum to 1 and have mean 1
     idx[axis] = 1
-    assert np.allclose(dsync[idx], 1)
+    assert np.allclose(dsync[tuple(idx)], 1)
 
 
 @pytest.mark.parametrize("aggregate", [np.mean, np.max])

